docs/src/apireference.md

```diff
@@ -547,6 +547,43 @@ Utilities.load
 Utilities.load_constraint
 ```
 
+### Caching optimizer
+
+Some solvers do not support incremental definition of optimization
+models. Nevertheless, you are still able to build incrementally an optimization
+model with such solvers. MathOptInterface allows to use a utility,
+[`Utilities.CachingOptimizer`](@ref), that will store in a [`ModelLike`](@ref) `model_cache`
+the optimization model during its incremental definition. Once the
+model completely defined, the `CachingOptimizer` specifies all problem's
+information to the underlying solver, all in once.
+
+The way to operate a `CachingOptimizer` is as follow.
+1) Suppose you want to cache a (empty) [`AbstractOptimizer`](@ref) `optimizer`.
+2) Define a [`ModelLike`](@ref) `model_cache` that will store the incremental definition of the model.
+3) Cache `optimizer` with `model_cache` by calling `CachingOptimizer(model_cache, optimizer)`. This method returns an empty `AbstractOptimizer` `cached_optimizer`.
```

Comment:
There is also a constructor where you only give the cache and the mode and then to give the optimizer with `attach_optimizer`.